presto-hive/src/main/java/com/facebook/presto/hive/GenericHiveRecordCursorProvider.java

```diff
@@ -72,16 +72,20 @@ public GenericHiveRecordCursorProvider(HdfsEnvironment hdfsEnvironment)
 
         RecordReader<?, ?> recordReader = hdfsEnvironment.doAs(session.getUser(),
                 () -> HiveUtil.createRecordReader(configuration, path, start, length, schema, columns, customSplitInfo));
-
-        return Optional.of(new GenericHiveRecordCursor<>(
-                configuration,
-                path,
-                genericRecordReader(recordReader),
-                length,
-                schema,
-                columns,
-                hiveStorageTimeZone,
-                typeManager));
+        final Optional<RecordCursor>[] recordCursors = new Optional[]{Optional.empty()};
+        hdfsEnvironment.doAs(session.getUser(),
+                () -> {
+            recordCursors[0] = Optional.of(new GenericHiveRecordCursor<>(
+                    configuration,
+                    path,
+                    genericRecordReader(recordReader),
+                    length,
+                    schema,
+                    columns,
+                    hiveStorageTimeZone,
+                    typeManager));
+                });
+        return recordCursors[0];
```

Comment:
nit: you can return the value returned by call `hdfsEnvironment.doAs(...)` like `return hdfsEnvironment.doAs`

Reply:
thanks! updated